### PR TITLE
Move navigator's ApiCodecCompressed to new lf-value-json library

### PR DESCRIPTION
**Pull request by S11001001**
Move the `ApiCodecCompressed` from navigator backend code to the new `//ledger-service/lf-value-json` library, continuing the plan for a LF value type-sensitive JSON encoding described in #2053, to make something useful for #1994 / #1628.

* [x] merge #2053 first

### Pull Request Checklist

- [x] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/master/CONTRIBUTING.md)
- [x] Include appropriate tests
- [x] Set a descriptive title and thorough description
- [x] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [x] Add a line to the [release notes](https://github.com/digital-asset/daml/blob/master/unreleased.rst), if appropriate

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: daml-lf/transaction-scalacheck/src/main/scala/com/digitalasset/daml/lf/value/TypedValueGenerators.scala [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+  * generate the type `[Map Decimal]`, which would be accompanied by a `Gen` that produced
+  * [[Value]]s each guaranteed to be a list of maps, whose values are all guaranteed to
+  * be Decimals.
+  *
+  * As a user, you will probably be most interested in one of the generators derived
+  * from `genAddend`; if you need a generator in this theme not already supported by
+  * one such generator, you can probably derive a new one from `genAddend` yourself.
+  */
+object TypedValueGenerators {
+  sealed abstract class ValueAddend {
```
Comment (S11001001) [inline daml-lf/transaction-scalacheck/src/main/scala/com/digitalasset/daml/lf/value/TypedValueGenerators.scala]: The name "addend" refers to how values of the type will be summed with other values to form a description of possible Values.

File: daml-lf/transaction-scalacheck/src/main/scala/com/digitalasset/daml/lf/value/TypedValueGenerators.scala [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+    import Value._, ValueGenerators.Implicits._
+    val text = noCid(PT.Text, ValueText) { case ValueText(t) => t }
+    val int64 = noCid(PT.Int64, ValueInt64) { case ValueInt64(i) => i }
+    val decimal = noCid(PT.Decimal, ValueDecimal) { case ValueDecimal(d) => d }
+    val unit = noCid(PT.Unit, (_: Unit) => ValueUnit) { case ValueUnit => () }
+    val date = noCid(PT.Date, ValueDate) { case ValueDate(d) => d }
+    val timestamp = noCid(PT.Timestamp, ValueTimestamp) { case ValueTimestamp(t) => t }
+    val bool = noCid(PT.Bool, ValueBool) { case ValueBool(b) => b }
+    val party = noCid(PT.Party, ValueParty) { case ValueParty(p) => p }
+
```
Comment (leo-da) [inline daml-lf/transaction-scalacheck/src/main/scala/com/digitalasset/daml/lf/value/TypedValueGenerators.scala]: what about `PT.List` and `PT.Optional`

Comment (leo-da) [reply daml-lf/transaction-scalacheck/src/main/scala/com/digitalasset/daml/lf/value/TypedValueGenerators.scala]: and also `PT.Map`

Comment (S11001001) [reply daml-lf/transaction-scalacheck/src/main/scala/com/digitalasset/daml/lf/value/TypedValueGenerators.scala]: Not possible to do with `noCid`, scroll down for support for those three, and also contractId.

File: ledger-service/lf-value-json/src/test/scala/com/digitalasset/daml/lf/value/json/ApiCodecCompressedSpec.scala [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+import org.scalatest.{Matchers, WordSpec}
+import org.scalatest.prop.GeneratorDrivenPropertyChecks
+import org.scalacheck.{Arbitrary, Gen}
+
+import scala.util.{Success, Try}
+
+class ApiCodecCompressedSpec extends WordSpec with Matchers with GeneratorDrivenPropertyChecks {
+
+  /** XXX SC replace when TypedValueGenerators supports TypeCons */
+  private val typeLookup: NavigatorModelAliases.DamlLfTypeLookup = _ => None
```
[the remark this answers is not in the document]
Comment (S11001001) [reply ledger-service/lf-value-json/src/test/scala/com/digitalasset/daml/lf/value/json/ApiCodecCompressedSpec.scala]: Not when we have records/variants/enums being produced by TypedValueGenerators, but for now it's fine.

File: daml-lf/transaction-scalacheck/src/main/scala/com/digitalasset/daml/lf/value/TypedValueGenerators.scala (hunk 1 above)
Comment (S11001001) [reply daml-lf/transaction-scalacheck/src/main/scala/com/digitalasset/daml/lf/value/TypedValueGenerators.scala]: Projection

**Pull request merged**